Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/organizer/submission/server/OrganizerSubmissionServer.kt

```diff
@@ -0,0 +1,67 @@
+package de.rki.coronawarnapp.presencetracing.organizer.submission.server
+
+import com.google.protobuf.ByteString
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.presencetracing.checkins.CheckInsReport
+import de.rki.coronawarnapp.server.protocols.internal.SubmissionPayloadOuterClass.SubmissionPayload
+import de.rki.coronawarnapp.util.PaddingTool
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import javax.inject.Inject
+
+class OrganizerSubmissionServer @Inject constructor(
+    private val paddingTool: PaddingTool,
+    private val appConfigProvider: AppConfigProvider,
+    private val dispatcherProvider: DispatcherProvider
+) {
+    suspend fun submit(
+        uploadTAN: String,
+        checkInsReport: CheckInsReport
+    ) = withContext(dispatcherProvider.IO) {
+        Timber.tag(TAG).d("submit(uploadTAN=%s, checkInReport=%s)", uploadTAN, checkInsReport)
+        val plausibleParameters = appConfigProvider
+            .getAppConfig()
+            .presenceTracing
+            .plausibleDeniabilityParameters
+
+        val checkInPadding = paddingTool.checkInPadding(
+            plausibleParameters = plausibleParameters,
+            checkInListSize = checkInsReport.encryptedCheckIns.size
+        )
+
+        val submissionPayload = SubmissionPayload.newBuilder()
+            .addAllKeys(emptyList())
+            .setRequestPadding(ByteString.copyFromUtf8(checkInPadding))
+            .setConsentToFederation(false)
+            .addAllVisitedCountries(emptyList())
+            .addAllCheckIns(checkInsReport.unencryptedCheckIns)
+            .addAllCheckInProtectedReports(checkInsReport.encryptedCheckIns)
+            .setSubmissionType(SubmissionPayload.SubmissionType.SUBMISSION_TYPE_HOST_WARNING)
+            .build()
+
+        // TODO submit
+    }
+
+    suspend fun submitFakePayload() = withContext(dispatcherProvider.IO) {
```

Comment:
Just asking for my PR:
Do you need some special header for fake submissions like cwa-fake?

Reply:
Good question : @mlenkeit  what do you think? do we need a special header here ? or do empty payload is enough?

Reply:
I would suggest to send the fake submission exactly the same way like for the regular submission, even to the same server endpoint (`/diagnosis-keys`) and with `cwa-fake`

Reply:
Done